lib/geo_concerns/processors/ogr_projection.rb

```diff
@@ -0,0 +1,44 @@
+require 'open-uri'
+
+module GeoConcerns::Processors
+  class OgrProjection < Hydra::Derivatives::Processors::Processor
+    include Hydra::Derivatives::Processors::ShellBasedProcessor
+    attr_reader :source_path, :directives
+
+    def self.process(source_path, directives)
+      new(source_path, directives).reproject # XXX: clean up after ourselves
+    end
+
+    def initialize(source_path, directives)
+      @source_path = source_path
+      @directives = directives
+
+      @output_filename = output_file(File.extname(source_path).gsub(/^\./, ''))
+      @output_directory = File.dirname(@output_filename)
+
+      FileUtils.mkdir_p(@output_directory) unless File.directory?(@output_directory)
+    end
+
+    def reproject
+      ogr2ogr
+      normalize_prj_file
+    end
+
+    def prettywkt
+      @srid ||= directives.fetch(:srid, 4326)
+      @prettywkt ||= open("http://spatialreference.org/ref/epsg/#{@srid}/prettywkt/").read
+    end
+
+    # reproject via OGR, @see http://www.gdal.org/ogr2ogr.html
+    # prevent recoding using SHAPE_ENCODING environment variable
+    def ogr2ogr
+      cmd = "env SHAPE_ENCODING= ogr2ogr -q -t_srs '#{prettywkt}' '#{@output_filename}' '#{source_path}'"
```

Comment:
Should there be an option to force the preservation of FID's during the reprojection?  This may be unnecessary given that this is the default behavior for more recent releases of the GDAL, but I am uncertain as to whether or not GeoConcerns ensures that the host environment has installed releases above a certain version:
> Use the FID of the source features instead of letting the output driver to automatically assign a new one. Note: starting with GDAL 2.0, if not in append mode, this behaviour becomes the default if the output driver has a FID layer creation option. In which case the name of the source FID column will be used and source feature IDs will be attempted to be preserved. This behaviour can be disabled by setting -unsetFid 
(http://www.gdal.org/ogr2ogr.html)

Please disregard this comment if this is not likely to be encountered within older server environments.